nifi-extension-bundles/nifi-github-bundle/nifi-github-extensions/src/main/java/org/apache/nifi/github/GitHubRepositoryClient.java

```diff
@@ -74,6 +81,15 @@ private GitHubRepositoryClient(final Builder builder) throws IOException, FlowRe
         switch (authenticationType) {
             case PERSONAL_ACCESS_TOKEN -> gitHubBuilder.withOAuthToken(builder.personalAccessToken);
             case APP_INSTALLATION_TOKEN -> gitHubBuilder.withAppInstallationToken(builder.appInstallationToken);
+            case APP_INSTALLATION_ID_AND_PRIVATE_KEY -> {
+                try {
+                    JWTTokenProvider jwtTokenProvider = new JWTTokenProvider(builder().appId, builder().privateKey);
+                    String token = jwtTokenProvider.getEncodedAuthorization();
+                    gitHubBuilder.withJwtToken(token);
+                } catch (Exception e) {
+                    throw new FlowRegistryException(e.getMessage());
```

Comment:
A custom message should be set, and the exception passed to the constructor to preserve the stack trace.
```suggestion
                    throw new FlowRegistryException("Failed to generate JWT from App ID and Private Key", e);
```